antarest/study/storage/utils.py

```diff
@@ -325,12 +350,11 @@ def get_start_date(
             steps = (13 - start_date.month) + end_date.month
 
     first_week_offset = 0
-    while True:
-        if (start_date + timedelta(days=first_week_offset)).strftime(
-            "%A"
-        ) == first_week_day:
+    for first_week_offset in range(7):
+        first_day = start_date + timedelta(days=first_week_offset)
+        first_day_name = ENGLISH_DAY_NAMES[first_day.weekday()]
+        if first_day_name == first_week_day.title():
```

Comment:
Same as above about weekday comparison: would be clearer based on integer representation, in my opinion.
Actually here we could just substract the 2 weekdays to get the offset, I guess,with an additional `% 7`. Not sure it's clearer though. We could have a small named method to clarify.

Reply:
The algorithm is based on searching a mathing day name (not day offset).
What do you mean? Any suggestion?